Fix Verizon Straight Talk Conflict add MMS

curiouscrusher opened the pull request:
## Changes

- I noticed when saving my SMS via Email notification settings that any recurring visits to the settings page showed "Straight Talk" instead of "Verizon" in the drop down. Turns out they each have the same SMS Email address, so I consolidated both Verizon & Straight Talk into single dropdown option. 
- Verizon SMS Email sends a traditional SMS which has a character limit. Between the subject line, console/store/date/credit info I wasn't even getting half the url to the store page in my text. After some quick testing I found Verizon supports MMS messaging via `number@vzwpix.com`. I tested and confirmed this sent successfully and included all the information, most importantly the entire product url. It appears there are also MMS addresses for AT&T and other major carriers, unfortunately I can only test Verizon currently. 
- My IDE cleaned up a bunch of extra whitespaces in the `index.html`

Prince25 commented on server/index.html:
```diff
@@ -1151,11 +1151,11 @@
 							"Ting": "message.ting.com",
 							"U.S. Cellular": "email.uscc.net",
-							"Verizon Wireless": "vtext.com",
+							"Verizon Wireless / Straight Talk": "vtext.com",
```
I believe Verizon Wireless and Straight Talk should be separate just because it's less work for a user to find Straight Talk alphabetically rather than to look for Verizon Wireless. They might not know that Verizon Wireless and Straight Talk have the same SMS email address. What do you think?

curiouscrusher replied:
This is a very valid point @Prince25.. On the one hand it does make sense to have the carriers listed separately since they are technically two different carriers. However it's also confusing when you go to the settings page and see a different carrier shown in the dropdown than what you selected previously. Perhaps something that checks both the carrier name and address when reading from the config to populate the dropdown, or there could be a note either in the docs or front-end that outlines some carriers share addresses and as such are grouped together in the dropdown. 

Thoughts?

Prince25 replied:
You, my friend, found a bug I wasn't aware of! The way it works is that it tries to match the SMS email address that's currently in `config.json` to the carrier. Since Straight Talk and Verizon Wireless have the same SMS email address, it matches with Straight Talk whether you chose Straight Talk or Verizon Wireless since Straight Talk comes first on the list. The same bug exists for AT&T and FirstNet. 

Your changes would solve this problem. I currently can't think of a simple way to solve this issue while keeping them separate. So for now, could you also group AT&T and FirstNet as `AT&T / FirstNet`?

Prince25 reviewed (changes requested):
Hello Nate,

Thanks for your detailed PR and for your changes. I apologize that you had through the atrocious (on purpose) `index.html`. I think your changes are fine, however, I did leave one suggestion.

Is it possible for you to find the MMS addresses for other carriers and add them as well? For testing, we can ask friends and even rely on users.

Thank you!

curiouscrusher commented:
Hi @Prince25, thanks for reviewing! I can go through and dig up some other carrier MMS addresses and add them in here for sure, will try and do that this weekend. 

Cheers

curiouscrusher commented:
As promised @Prince25, pushed those changes! 
I consolidated all carriers that share an address, added all the additional MMS addresses I could find, and updated the tooltip with some notes about the SMS vs MMS sending.

Prince25 commented:
Thank you!! It looks good to me. Have you seen what the longer Carrier names look like on the settings page?

One final thing would be to update the Email part of the SMS section of the README. Could you just update the carriers and perhaps briefly explain the difference between SMS and MMS? Appreciate it.
https://github.com/curiouscrusher/StockAlertBot/blob/b59b7678b6c2e8b7f86a2a2a9ddbfe7b1aecdc26/README.md?plain=1#L173

curiouscrusher commented:
Hey @Prince25, sorry for the delay! Holiday madness set in and I've been slammed. 

I'm guessing the longer carrier names don't look so great in the Material dropdowns? I'll have some downtime for the holidays starting next week, I can take a look at the visual for the dropdowns and write up some explanation on MMS/SMS too. 

Cheers!
